### PR TITLE
Chandra/onboarding feature

### PR DESCRIPTION
**Pull request by curesh**
Onboarding PR

### CODE REVIEW

File: cs_todo.txt
Comment (ryanmtran) [inline cs_todo.txt]: Remove this file from commit history

File: src/assets/images/OnboardingFruit1.svg
Comment (ryanmtran) [inline src/assets/images/OnboardingFruit1.svg]: Change import to reference OnboardingFruit on signup screen

File: src/assets/images/OnboardingFruit2.svg
Comment (ryanmtran) [inline src/assets/images/OnboardingFruit2.svg]: Change import to reference OnboardingFruit on signup screen

File: src/components/Onboarding/Fruit1.svg
Comment (ryanmtran) [inline src/components/Onboarding/Fruit1.svg]: Delete this file

File: src/components/Onboarding/Onboarding.css
Comment (ryanmtran) [inline src/components/Onboarding/Onboarding.css]: Delete this

File: src/components/Onboarding/ProgressBar.js
Comment (ryanmtran) [inline src/components/Onboarding/ProgressBar.js]: Delete this file (because you mentioned it wasn't part of your code)

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js:135]: Delete console.log

File: src/components/SignUp/SignUp.css
Comment (ryanmtran) [inline src/components/SignUp/SignUp.css]: I think once we get this all figured out, you can prolly delete SignUp.css and the previous SignUpScreen from our repo

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Let's change the name of this component to SignUpScreen and the directory to be SignUp

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Take this comment out

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Copy this css from our figma:
background: 'FFFFFF',
borderRadius: '15px',
boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',

Also, stretch goal, try to remove the current hover and make it match the hover of our Figma prototype

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Can we put this next to ROLE_NAMES? Also, is there a way for us to choose one or the other? I know that there might be conflicts with the database schema and figma, so totally cool if we can't

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: These Fruit imports are gonna change

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Ooooh this is interesting, what is useStyles.fruitNum doing? are you creating a new useStyles? what's the benefit of this as opposed to making a fruit3 selector in the classes instantiation of useStyles above? Also, nitpick but why are we going fruit 3 --> fruit2 --> fruit1? Can we make the id's match the fruitNum? (Totally cool if we have to call it Fruit0, Fruit1, Fruit2

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Same as above

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Same as above

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Remove console.logs

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: What's the point of >=3?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: I'm curious as to why we make role an integer? Could we use the  index + ROLE_NAME or ROLE_TYPES array to make it a string? Just an idea

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Can you implement verifications of valid emails and phones?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Can you check zipcode length as well? I'm pretty sure it's 5? Double check me on that

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: zipcode is repeated twice

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: I'm pretty sure role 0 corresponds to the buyer, but why would they be redirected to sign in? Aren't they the same where you have to be authenticated before you can sign in?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Remove this! You're currently rendering the role number client-side

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: This handleSubmit function should also prolly check if a user account already exists, and send out an error accordingly! Also, we need to make sure that they can't go to the final stage of our registration process if that's the case

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Question in general, can we implement a popup so that way when the user presses next at step 3, there is a confirmation message? Don't worry about implementing this now, I just messaged the designers. For now, put a TODO comment in your code to make note of that

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Can you add a comment that defines the mappings of role numbers to user role types?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Let's rename this to something more connotative like isMatchingPasswordInputs or something

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Doesn't this implement your isInvalid above? can we combine the two?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Can we order the steps, Step1, Step2 ... in our code?

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Why does the submit button take us to the previous step? Let's implement it so that way it redirects us to the landing screen using history.push() --> I have this code in my Navbar and SignInScreen

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Same as above, do we need to call getSteps(), or can we just have a global array

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: I know this is a lot of comments, so this can definitely be a stretch goal, but can you refactor your code so that all of your Steps are in their own files? That would organize the code a lot!

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [inline src/components/Onboarding/OnboardingScreen.js]: Is there a way to remove the asterisk next to comments in the text input client side? I'm looking at the web app rn

**Review by ryanmtran — CHANGES_REQUESTED** [review]
WOOOO!!! Thank you for your hard work on this Chandra. This feature honestly looks beautiful, and I'm so excited for this to be integrated into our app. This was a really big feature, so there are a lot of comments, so feel free to lmk if you have any follow up questions or confusions.

File: src/components/Onboarding/OnboardingScreen.js
Comment (curesh) [reply src/components/Onboarding/OnboardingScreen.js]: Yeah, it just comes down to the differences between the schema and figma. I can change the schema to match figma though, if you'd prefer that

File: src/components/Onboarding/OnboardingScreen.js
Comment (curesh) [reply src/components/Onboarding/OnboardingScreen.js]: Thats a good catch! The "styling: useSty..." stuff is actually redundant bc of the fruit: { ... } css right below that. The reason why I have the css in that js array tho instead of in the makeStyles above, is bc this makes it easier when iterating through the buttons in the for loop below. I just need to call bt.fruit instead of having an if, elseif, else clause.

As for the the names of the fruits being backwards, yeah that is annoying loll. I copied the naming convention used in the figma tho, (where the left most fruit corresponded to Fruit3, and so on).

File: src/components/Onboarding/OnboardingScreen.js
Comment (curesh) [reply src/components/Onboarding/OnboardingScreen.js]: I just had it >= instead of == in case there was some lag or something and they managed to press the button twice. I wanted to make sure there was no undefined behavior

File: src/components/Onboarding/OnboardingScreen.js
Comment (curesh) [reply src/components/Onboarding/OnboardingScreen.js]: I just did it as an int cause this variable is used heavily in the html portion to compare to different roles, and I thought that doing "role == 1" might be more succinct. Would the benefit of the string version be code readability (cause I agree)?

File: src/components/Onboarding/OnboardingScreen.js
Comment (curesh) [reply src/components/Onboarding/OnboardingScreen.js]: Kind of; isInvalid returns a boolean depending on whether every single field is filled, and the repeatPwdCheck returns an error message based only on whether or not the password and repeat password match each other. I think they're different enough where I'm not sure how they would be combined

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js:136]: I just realized that for buyers, they only have to go through one authentication process. Can you do a check to see if the role is buyer, and if so, make account approval default to "unapproved" and registration approval to default as 
"approved"? That way F2P only has to approve them once.

**Review by ryanmtran — COMMENTED** [review]
> WOOOO!!! Thank you for your hard work on this Chandra. This feature honestly looks beautiful, and I'm so excited for this to be integrated into our app. This was a really big feature, so there are a lot of comments, so feel free to lmk if you have any follow up questions or confusions.

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [reply src/components/Onboarding/OnboardingScreen.js]: Honestly we can just keep it tbh because it might break a lot of features hahah

File: src/components/Onboarding/OnboardingScreen.js
Comment (ryanmtran) [reply src/components/Onboarding/OnboardingScreen.js]: I think it could benefit if we indexed into the array, but for the sake of this really long code review, you could probably just add a comment right above where you instantiate your useState for role and say something along the lines of... Role state corresponds with an integer that uses the following mapping : {1: ..., 2: ..., 3:...}

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Really minor nitpick, but we can just change the names of the fruits to be fruit0, fruit1, and fruit2 so that way we match the id. That shouldn't be a big deal if we change the name of the svg file! (So change the names to OnboardingFruit0.svg, OnboardingFruit1.svg, and OnboardingFruit2.svg, with the ordering that matches the buyer, agency, vendor order

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Remove this

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js:104]: Ooh you included this! Ignore my reply I made in github

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js:153]: Just set the errState and we can remove console logs

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Remove console log

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Ooooh cool! I figured this login stuff for selecting a different email! Refer to my slack messages to get that done and lmk if you have any questions

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Change this to be more connotative like: generateErrorMsg

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Rename this to be more connotative, like step3EmptyInputCheck

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Rename to something more connotative like validPhoneAndZipcodeCheck

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Remove comments for PRs that were used for testing

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: Same as before is there a reason why we can't order the steps?

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js:1]: For your signup screen notice how the text field values bleed through the side navbar? (When your window is small and click the navbar extension icon in the upper right part of the screen). To fix that, you can change the z-index for text inputs to 0. Refer to my newly pushed SignInScreen if you need help with that!

File: src/components/SignUp/SignUpScreen.js
Comment (ryanmtran) [inline src/components/SignUp/SignUpScreen.js]: When you press Done, don't forget to do history.push() to take you to the landing page!

**Review by ryanmtran — CHANGES_REQUESTED** [review]
Hey Chandra! This feature is fire. Just some changes requested for some naming conventions, using history.push() when you press the button at step4, inclusion of the login user already exists (where I slack messaged the way to go about it!), and some other fixes! Also, when you finish incorporating these changes, don't forget to do user testing with your feature and try to hone in on edge cases and user pathways. I can't wait to get this merged and thank you so much for your hard work.

**Review by ryanmtran — APPROVED** [review]
This looks amazing chandra, and the fact that you were able to clean up the code so well I'm so impressed with this feature! This looks great to me, and I feel super happy merging this into master. As a note so we don't forget, the next steps in elevating this feature would be styling the error message and creating a popup that asks for confirmation when the user presses finish (We can borrow that from Diya's feature probably!)

I'm so excited Chandra, and I can't wait for this to get merged. Woo!!!

**Pull request merged**
